dialer.go

```diff
@@ -275,6 +294,10 @@ func (d *Dialer) Dial(ctx context.Context, instance string, opts ...DialOption)
 	}), nil
 }
 
+func invalidClientCert(c *tls.Config) bool {
+	return time.Now().After(c.Certificates[0].Leaf.NotAfter)
```

Comment:
What happens if some part of this statement `c.Certificates[0].Leaf.NotAfter` is nil? Can we guarantee that whenever this function is called, each of these fields will have a value: `c.Certificates`, `c.Certificates[0]`, `c.Certificates[0].Leaf`, `c.Certificates[0].Leaf.NotAfter`?

Reply:
This same line of code is in v1 proxy, so I think we're safe. However, it's cheap to add those intermediate checks. I'll do that.

Reply:
Fixed.